code/modules/research/xenobiology/crossbreeding/regenerative.dm

```diff
@@ -23,13 +23,24 @@ Regenerative extracts:
 		to_chat(user, "<span class='warning'>[src] will not work on the dead!</span>")
 		return
 	if(H != user)
-		user.visible_message("<span class='notice'>[user] crushes the [src] over [H], the milky goo quickly regenerating all of [H.p_their()] injuries!</span>",
-			"<span class='notice'>You squeeze the [src], and it bursts over [H], the milky goo regenerating [H.p_their()] injuries.</span>")
+		if(!do_mob(user, H, 10)) // 1 second delay
+			return FALSE
+		user.visible_message("<span class='notice'>[user] crushes the [src] over [H], the milky goo quickly regenerating some of [H.p_their()] injuries!</span>",
+			"<span class='notice'>You squeeze the [src], and it bursts over [H], the milky goo regenerating some of [H.p_their()] injuries.</span>")
 	else
-		user.visible_message("<span class='notice'>[user] crushes the [src] over [user.p_them()]self, the milky goo quickly regenerating all of [user.p_their()] injuries!</span>",
-			"<span class='notice'>You squeeze the [src], and it bursts in your hand, splashing you with milky goo which quickly regenerates your injuries!</span>")
+		if(!do_mob(user, H, 20)) // 2 second delay
+			return FALSE
+		user.visible_message("<span class='notice'>[user] crushes the [src] over [user.p_them()]self, the milky goo quickly regenerating some of [user.p_their()] injuries!</span>",
+			"<span class='notice'>You squeeze the [src], and it bursts in your hand, splashing you with milky goo which quickly regenerates som of your injuries!</span>")
 	core_effect_before(H, user)
-	H.revive(full_heal = TRUE, admin_revive = FALSE)
```

Comment:
There's still a healing over time effect as it injects some regenerative jelly. The fact this was a full, instant heal is the main issue. Especially for something you can just pack en masse in boxes.

With the current formula, if you take 100 damage. You will heal 40, then another 18.75 as the reagents are consumed. Or 58.75 of all common damage types in total. That's quite a lot of damage to heal for one medical item. And these aren't that incredibly hard to mass produce. 

I'll adjust this to 15 + 40% + 7.5 jelly.  That way when you take 100 damage, you'll heal 55, then another 28.125 health overtime. Which is 83.125 in total. 

Because of the way cores are coded, I can make different cores better/worse at healing certain damage types, including one to heal bones specifically. I'll go ahead and do that.

Reply:
has this been addressed

Reply:
Partially. I've raised it to 10 + 40% and increased the jelly. If you take 100 damage it should heal like 85% which is roughly the same effect as wanted by vexy but less instant. I don't mind raising it further but I feel then they might get too strong again. Also uh... stuff like the adamantine jello heal more . So I'd say it's pretty much fully addressed.

Reply:
Adamantine is like 10 + 65% but specifically for brute. For example.

Reply:
yeah sure i'll call that resolved